src/main/flight/imu.c

```diff
@@ -529,3 +526,73 @@ void imuSetHasNewData(uint32_t dt)
 }
 #endif
 
+void imuQuaternionComputeProducts(quaternion *quat) {
+    quat->ww = quat->w * quat->w;
+    quat->wx = quat->w * quat->x;
+    quat->wy = quat->w * quat->y;
+    quat->wz = quat->w * quat->z;
+    quat->xx = quat->x * quat->x;
+    quat->xy = quat->x * quat->y;
+    quat->xz = quat->x * quat->z;
+    quat->yy = quat->y * quat->y;
+    quat->yz = quat->y * quat->z;
+    quat->zz = quat->z * quat->z;
+}
+
+bool imuQuaternionHeadfreeDislocationSet(void) {
```

Comment:
`imuHeadfreeSetReference` would be much nicer.
"Dislocation" is indeed a strange choice of wording for this.